R/article.R

```diff
@@ -281,19 +281,67 @@ mnras_article <- function(..., keep_tex = TRUE, fig_caption = TRUE) {
   )
 }
 
+#' @param document_style For \code{oup_article}, one of "contemporary", "modern", or "traditional" setting overall style of document. Defaults to "contemporary".
+#' @param papersize For \code{oup_article}, one of "large", "medium", or "small" setting output page size. Defaults to "large".
+#' @param namedate For \code{oup_article}, a logical variable indicating if natbib citations should be in name-date format. Defaults to \code{FALSE}.
+#' @param onecolumn For \code{oup_article}, a logical variable indicating if one column formatting should be used. Defaults to \code{FALSE}.
 #' @section \code{oup_article}: Format for creating submissions to many Oxford University Press
 #'   journals. Adapted from
 #'   \url{https://academic.oup.com/journals/pages/authors/preparing_your_manuscript}
-#'   and \url{https://academic.oup.com/icesjms/pages/General_Instructions}.
+#'   and the \code{oup-authoring-template} available on CTAN at
+#'   \url{https://www.ctan.org/pkg/oup-authoring-template}.
 #' @export
 #' @rdname article
 oup_article <- function(
   ..., keep_tex = TRUE,
-  md_extensions = c("-autolink_bare_uris")
+  md_extensions = c("-autolink_bare_uris"),
+  journal=NULL,pandoc_args=NULL,
+  number_sections=TRUE,
+  citation_package=c("natbib","default"),
+  papersize=c("large","medium","small"),
+  document_style=c("contemporary","modern","traditional"),
+  namedate=FALSE,onecolumn=FALSE
 ) {
+  citation_package <- match.arg(citation_package)
+  if (length(citation_package) > 1) stop("Please choose just one ", dQuote("citation_package"))
+  papersize <- match.arg(papersize)
+  if (length(papersize) > 1) stop("Please choose just one ", dQuote("papersize"))
+  document_style <- match.arg(document_style)
+  if (length(document_style) > 1) stop("Please choose just one ", dQuote("document_style"))
+
+  args <- c(
+    journal = journal,
+    papersize = papersize,
+    document_style = document_style
+  )
+
+  # Convert to pandoc arguments
+  pandoc_arg_list <- mapply(rmarkdown::pandoc_variable_arg, names(args), args,
+                            SIMPLIFY = FALSE, USE.NAMES = FALSE)
+
+  # namedate
+  if (namedate)
+    pandoc_arg_list = c(pandoc_arg_list,
+                        rmarkdown::pandoc_variable_arg("namedate"))
+
+  # onecolumn
+  if (onecolumn)
+    pandoc_arg_list = c(pandoc_arg_list,
+                        rmarkdown::pandoc_variable_arg("onecolumn"))
+
+  # section numbering
+  if (!number_sections)
+    pandoc_arg_list = c(pandoc_arg_list,
+                        rmarkdown::pandoc_variable_arg("unnumsec"))
+
+  pandoc_arg_list <- unlist(pandoc_arg_list)
```

Comment:
I didn't previously see these messages for some reason. 

The reason I placed these in the format function is that this way you can quickly change between article formats (for example, if the article is rejected in one journal, you can quickly switch to another). This is also why I have tried to follow the overall skeleton/YAML of Elsevier articles. In general, I tried to place all things that are likely to be specific to the OUP format in the formatting function. You can still, however, put them at the base level, it just makes the Rmd file a bit less portable to another journal. In my opinion, this is how all journal formats should do it and is more consistent with how the base rmarkdown format works (e.g., `pdf_document` has the `citation_package` and `latex_engine` arguments, but `html_document` does not, and these arguments are not base YAML entries). 

One example of why I did this: the oup cls file has specific non-standard definitions of what `papersize` means. If I left it as a base YAML entry, then when you switch to another journal, it will fail or do strange things as that other journal will likely call `a4` what OUP calls `large`.

Reply:
That are good points. 

Currently **rticles** formats and template are not designed with switching the format function in the template. That is why the base level variable in metadata are used a lot. This is also because it is easy and avoid having a logic inside a format to pass them as variable. The command line will be rather long. but we can live with that. 

So I understand the idea. I'll think about it because maybe it is a better way to deal will all the templates. 

Currently we are more trying to reach for an uniformisation of the variables names and same kind of YAML header for all template. 

> and is more consistent with how the base rmarkdown format works (e.g., pdf_document has the citation_package and latex_engine arguments, but html_document does not, and these arguments are not base YAML entries).

About this, we do that mainly because we need to know the value for some processing, and we can even modified it or pass other option. So basically, they are not passed directly as pandoc variables. For any variable that can be passed to Pandoc directly as Pandoc variables without any more logic, then the usual metadata variable syntax is used.

Reply:
Yes, I get that `citation_package` et al. have a somewhat different role than `papersize`, but conceptually I think it makes more sense to put format specific options in the format function. If it is not hard to put the options in the format function, then I don't see why not to do it. This all goes along with my "generic article format" I proposed elsewhere. The idea would be to write the article once using the generic format and then just change the format function to get to that of a specific journal. This seems like a good long-term goal for rticles, though I understand this will require quite a bit of work.